matchers/matcher.go

```diff
@@ -289,6 +292,42 @@ func objectToString(obj interface{}) string {
 	}
 }
 
+type FieldMatchArgs struct {
+	name      string
+	matchType reflect.Type
+	params    params
+}
+type FieldStrategyFunc func(field reflect.StructField) FieldMatchArgs
+
+var DefaultFieldStrategyFunc = func(field reflect.StructField) FieldMatchArgs {
```

Comment:
extracted your original code as a default strategy and exported so i can fall back on this when needed